### PR TITLE
Improved testing

### LINKED ISSUE #13457: add helper function to testcase to execute inaccesable methods
instead of
```php
$reflector = new \ReflectionClass($this->service);
$method = $reflector->getMethod('filterMutedNotifications');
$method->setAccessible(true);

$result = $method->invoke($this->service, $subscriptions, $event);
```

make this work:
```php
$result = $this->executeInaccessableMethod([$this->service, 'filterMutedNotifications'], [$this->service, $subscriptions, $event]);
```

### LINKED ISSUE #14230: Add tests for ElggWidget::move
no test exist that test moving widgets

maybe also a a widget/move and/or widget/save action test that action test that a widget moved to the correct new position

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #14230
fixes #13457

### CODE REVIEW

File: engine/tests/phpunit/integration/Elgg/WidgetsServiceIntegrationTest.php [hunk 1]
```diff
@@ -8,7 +8,18 @@
  * Elgg Test \ElggWidget
  */
 class WidgetsServiceIntegrationTest extends IntegrationTestCase {
-
+	
+	public function testHandlerRegistration() {
+		$this->assertFalse(elgg_is_widget_type('test_handler'));
+		elgg_unregister_widget_type('test_handler');
```
Comment (jeabakker) [inline engine/tests/phpunit/integration/Elgg/WidgetsServiceIntegrationTest.php:14]: why?

File: engine/tests/phpunit/integration/ElggWidgetIntegrationTest.php [hunk 2]
```diff
@@ -67,15 +67,368 @@ public function testGetTitle() {
...
 	}
-	
-	public function testHandlerRegistration() {
-		$this->assertFalse(elgg_is_widget_type('test_handler'));
-		elgg_unregister_widget_type('test_handler');
 		
-		elgg_register_widget_type(['id' => 'test_handler', 'context' => []]);
-		$this->assertTrue(elgg_is_widget_type('test_handler'));
+	public function testMovingWidgetsWithinSameColumnToLastPosition() {
+		elgg_register_widget_type(['id' => 'widget_move_handler', 'context' => ['move_widget_context']]);
```
Comment (jeabakker) [inline engine/tests/phpunit/integration/ElggWidgetIntegrationTest.php:72]: this remains active during all the following tests???!!!

**Review by jeabakker — APPROVED** [review]

File: engine/tests/phpunit/integration/ElggWidgetIntegrationTest.php (hunk 2 above)
Comment (jdalsem) [reply engine/tests/phpunit/integration/ElggWidgetIntegrationTest.php]: now (un)registering in up/down

**Review by jeabakker — APPROVED** [review]

File: engine/tests/classes/Elgg/BaseTestCase.php [hunk 3]
```diff
@@ -331,5 +331,22 @@ public static function assertEquals($expected, $actual, $message = '', $delta =
...
+	public function invokeInaccessableMethod($argument, string $method, ...$args) {
+		$reflector = new \ReflectionClass($argument);
+		
+		$inaccessable_method = $reflector->getMethod($method);
+		$inaccessable_method->setAccessible(true);
+		
+		return $inaccessable_method->invoke($argument, ...$args);
+	}
+	
+	public function getInaccessableProperty($argument, string $property) {
```
Comment (jeabakker) [inline engine/tests/classes/Elgg/BaseTestCase.php]: why not ``protected``

**Review by jeabakker — CHANGES_REQUESTED** [review]

**Review by jeabakker — APPROVED** [review]

**Pull request merged**